alerts/cluster-etcd-operator/etcdGRPCRequestsSlow.md

```diff
@@ -73,24 +85,47 @@ To confirm this is the cause of the slow requests either:
       ))
 ```
 
+### Rogue Workloads
+
+In some cases, we've seen non-OpenShift workload put a lot of stress on the
+API server that eventually cascades into etcd. One specific instance was
+listing all pods across namespaces exhausting CPU and memory on API server
+and subsequently on etcd.
+
+Please consult the audit log and see whether some service accounts make
+suspicious calls, both in terms of generality (listings, many/all namespaces)
+and frequency (eg listing all pods every 10s).
+
+
 ## Mitigation
 
-### Fragmented state
+Depending on what resource was determined to be exhausted,
+you can try the following:
 
-In the case of slow disk or when the etcd DB size increases, we can defragment
-existing etcd DB to optimize DB consumption as described in
-[here][etcdDefragmentation]. Run the following command in all etcd pods.
+### CPU
+
+Find the offending process that uses too much CPU, try to limit or
+shutdown the process. If feasible on clouds,
+adding more or faster CPUs may help to reduce the latency.
+
+
+### Disk
+
+Find the offending process that causes the disk performance to degrade,
+this can also be a noisy neighbour process on the control plane node
+(eg fluentd with logging, OVN) or etcd itself. If the culprit is determined
+to etcd, try to reduce the load coming from the apiserver.
+Most commonly this also happens when a cluster is scaled up with
+many more nodes, so reducing the cluster scale again can help.
+
+If feasible on clouds, upgrading your storage or instance type
+can significantly increase your sequential IOPS and available bandwidth.
```

Comment:
The process for scaling up a control plane is still lacking official docs, isn't it?

Reply:
really? this is the first google hit: https://access.redhat.com/solutions/5377471